detection_rules/eswrap.py

```diff
@@ -384,3 +388,327 @@ def collect_events(ctx, host_id, query, index, rta_name, rule_id, view_events):
+
+    with open_model_file('dga_ngrams_create') as painless_install:
+        es_client.put_script(id='dga_ngrams_create', body=painless_install)
+        # f'{model_id}_dga_ngrams_create'
+
+    with open_model_file('dga_ngrams_transform_delete') as painless_delete:
+        es_client.put_script(id='dga_ngrams_transform_delete', body=painless_delete)
+        # f'{model_id}_dga_ngrams_transform_delete'
+
+    # Install ingest pipelines
+    click.secho('[+] Uploading pipelines')
+
+    def _build_es_script_error(err, pipeline_file):
+        error = err.info['error']
+        cause = error['caused_by']
+
+        error_msg = [
+            f'Script error while uploading {pipeline_file}: {cause["type"]} - {cause["reason"]}',
+            ' '.join(f'{k}: {v}' for k, v in error['position'].items()),
+            '\n'.join(error['script_stack'])
+        ]
+
+        return click.style('\n'.join(error_msg), fg='red')
+
+    with open_model_file('dns_enrich_pipeline') as ingest_pipeline1:
+        try:
+            ingest_client.put_pipeline(id='dns_enrich_pipeline', body=ingest_pipeline1)
+        except elasticsearch.RequestError as e:
+            if e.error == 'script_exception':
+                client_error(_build_es_script_error(e, 'ingest_pipeline1'), e, ctx=ctx)
+            else:
+                raise
+
+    with open_model_file('dns_dga_inference_enrich_pipeline') as ingest_pipeline2:
+        # try:
+        #     processors = ingest_pipeline2['processors']
+        #     inference_processor = next(p for p in processors if 'inference' in p)
+        #     inference_processor['inference']['model_id'] = model_id
+        # except StopIteration:
+        #     client_error(f'{get_model_filename("dga_*_ingest_pipeline2.json")} may be malformed - check file')
+        try:
+            ingest_client.put_pipeline(id='dns_dga_inference_enrich_pipeline', body=ingest_pipeline2)
+        except elasticsearch.RequestError as e:
+            if e.error == 'script_exception':
+                client_error(_build_es_script_error(e, 'ingest_pipeline2'), e, ctx=ctx)
+            else:
+                raise
+
+    click.echo('Ensure that you have updated your packetbeat.yml config file.')
+    click.echo('    - reference: ML_DGA.md #2-update-packetbeat-configuration')
+    click.echo('To upload rules, run: kibana upload-rule <dga-rule-files>')
+    click.echo('To upload ML jobs, run: es experimental upload-ml-job <dga-job-files>')
+
+
+@es_experimental.command('upload-ml-job')
+@click.argument('job-file', type=click.Path(exists=True, dir_okay=False))
+@click.option('--overwrite', '-o', is_flag=True, help='Overwrite job if exists by name')
+@click.pass_context
+def upload_ml_job(ctx: click.Context, job_file, overwrite):
+    """Upload experimental ML jobs."""
+    es_client: Elasticsearch = ctx.obj['es']
+    ml_client = MlClient(es_client)
+
+    with open(job_file, 'r') as f:
+        job = json.load(f)
```

Comment:
i feel like we should have `read_json`-like functions in utils.py by now

Reply:
true - I kind of wish json was stdlib by now and that pathlib had a `read_json` method